builder/amazon/common/step_security_group.go

```diff
@@ -45,7 +45,9 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 
 	port := s.CommConfig.Port()
 	if port == 0 {
-		panic("port must be set to a non-zero value.")
+		if s.CommConfig.Type != "none" {
+			panic("port must be set to a non-zero value.")
```

Comment:
although, when all is said and done here is anything supposed to happen on AWS?  Because I'm finding with this change that an ami still gets created.